src/store/aggregator/aggregator.sagas.ts

```diff
@@ -121,6 +128,58 @@ export function* fetchPausedTokens() {
   }
 }
 
+export function* fetchIncentive() {
+  try {
+
+    yield* put(aggregatorActions.setIncentivesLoading());
+
+    const rewardManager = yield* select(rewardManagerSelector);
+    const sendAmount = yield* select(sendAmountSelector);
+    const flowState = yield* select(flowStateSelector);
+
+    if (!rewardManager) {
+      throw new Error('Could not find RewardManager contract');
+    }
+
+    const startingTokenAddress = (yield* select(startingTokenAddressSelector)) ?? '';
+    const destinationTokenAddress = (yield* select(destinationTokenAddressSelector)) ?? '';
+
+    let incentive = BigNumber.from(0);
+    let receiveAmount = BigNumber.from(0);
+    let incentiveType = IncentiveType.none;
+
+    if(flowState === 'deposit' && startingTokenAddress) {
+
+      incentiveType = IncentiveType.reward;
+      const tokenDecimals = yield* select(startingTokenDecimalsSelector);
+      const amount = utils.parseUnits(sendAmount, tokenDecimals);
+      incentive = (yield* call(rewardManager.getRewardForDeposit,
+        startingTokenAddress.toLowerCase(), amount)) as BigNumber;
+      incentive = roundBN(incentive, 3);
```

Comment:
I don't think you should round `incentive`/`penalty` before you `add`/`subtract` them from `receiveAmount` because you'll lose some precision there.